src/StripNewlines.php

```diff
@@ -15,13 +16,18 @@ class StripNewlines extends AbstractFilter
      *
      * Returns $value without newline control characters
      *
-     * @param  string|array $value
-     * @return string|array
+     * @param  mixed $value
+     * @return mixed
      */
     public function filter($value)
     {
-        if (! is_scalar($value) && ! is_array($value)) {
-            return $value;
+        if (! is_array($value)) {
```

Comment:
Same question as above: what happens to non-scalars? Perhaps we need a small `@internal` utility class that does this `is_array()`, `is_scalar()` and `is_string()` check recursively, somehow? :thinking:

Reply:
name not important, but literally a common function (or static method) such as `apply_filter_only_to_stringable_and_stringable_array_values(mixed $value, callback $filter)`

Reply:
Do we want to do the same implicit casting to the array values?

If one of the array values contains an object, should the whole array be returned untouched? My only apprehension to this is BC. When considering how str_replace would have handled this previously:

Consider the scenario, you have a custom filter to change a collection of userIds into a collection of `User` objects, and that User class implements `__toString()`. Would `str_replace` cast these objects to a string implicitly previously?

This is a fairly contrived example I admit, but it demonstrates my thoughts regarding changing existing behaviour. This is starting to give me a headache :D.

Reply:
The reason why I asked about `is_array()` and recursion is that I don't know if that's what the `preg_replace()` functions do...